java/test/org/openqa/selenium/support/events/EventFiringDecoratorTest.java

```diff
@@ -86,31 +103,200 @@ public void beforeGet(WebDriver driver, String url) {
+          }
+          @Override
+          public void beforeFindElement(WebDriver driver, By locator) {
+            acc.append("beforeFindElement\n");
+          }
+
+          @Override
+          public void afterFindElement(WebDriver driver, By locator, WebElement result) {
+            acc.append("afterFindElement\n");
+          }
+          @Override
+          public void beforeFindElements(WebDriver driver, By locator) {
+            acc.append("beforeFindElements\n");
+          }
+
+          @Override
+          public void afterFindElements(WebDriver driver, By locator, List<WebElement> result) {
+            acc.append("afterFindElements\n");
+          }
+          @Override
+          public void beforeGetPageSource(WebDriver driver) {
+            acc.append("beforeGetPageSource\n");
+          }
+
+          @Override
+          public void afterGetPageSource(WebDriver driver, String result) {
+            acc.append("afterGetPageSource\n");
+          }
+
+          @Override
+          public void beforeClose(WebDriver driver) {
+            acc.append("beforeClose\n");
+          }
+
+          @Override
+          public void afterClose(WebDriver driver) {
+            acc.append("afterClose\n");
+          }
+
+          @Override
+          public void beforeQuit(WebDriver driver) {
+            acc.append("beforeQuit\n");
+          }
+
+          @Override
+          public void afterQuit(WebDriver driver) {
+            acc.append("afterQuit\n");
+          }
+
+          @Override
+          public void beforeGetWindowHandle(WebDriver driver) {
+            acc.append("beforeGetWindowHandle\n");
+          }
+
+          @Override
+          public void afterGetWindowHandle(WebDriver driver, String result) {
+            acc.append("afterGetWindowHandle\n");
+          }
+
+          @Override
+          public void beforeGetWindowHandles(WebDriver driver) {
+            acc.append("beforeGetWindowHandles\n");
+          }
+
+          @Override
+          public void afterGetWindowHandles(WebDriver driver, Set<String> result) {
+            acc.append("afterGetWindowHandles\n");
+          }
         };
     WebDriver decorated = new EventFiringDecorator<>(listener).decorate(driver);
 
     decorated.get("http://example.com/");
+    decorated.getCurrentUrl();
+    decorated.getTitle();
+    decorated.findElement(By.id("test"));
+    decorated.findElements(By.id("test"));
+    decorated.getPageSource();
+    decorated.getWindowHandle();
+    decorated.getWindowHandles();
+    decorated.close();
+    decorated.quit();
 
     assertThat(listener.acc.toString().trim())
         .isEqualTo(
             String.join(
                 "\n",
-                "beforeAnyCall get",
                 "beforeAnyWebDriverCall get",
                 "beforeGet",
                 "afterGet",
                 "afterAnyWebDriverCall get",
-                "afterAnyCall get"));
+                "beforeAnyWebDriverCall getCurrentUrl",
```

Comment:
I appreciate adding exhaustive tests. But it might be a good idea to split the PRs out. For example, removal could be one PR and one PR could be adding tests. It will help with cleaner commits.